src/material/datepicker/calendar-body.ts

```diff
@@ -467,6 +479,17 @@ export class MatCalendarBody<D = any> implements OnChanges, OnDestroy, AfterView
     } else if (this.endValue === value) {
       return this._endDateLabelId;
     }
+
+    if (this.comparisonStart !== null && this.comparisonEnd !== null) {
+      if (value === this.comparisonStart && value === this.comparisonEnd) {
+        return `${this._comparisonStartDateLabelId}-${this._comparisonEndDateLabelId}`;
```

Comment:
I'm still a little confused by the hyphen - isn't this supposed to be space-deliminated?

Reply:
Apologies for the typo again, I snagged the snippet from the wrong version. Updated in the latest refactor commit.